rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -705,3 +739,77 @@ def parse_service_string(pkg_name, srv_name, message_string):
         pkg_name, srv_name + SERVICE_RESPONSE_MESSAGE_SUFFIX, response_message_string)
 
     return ServiceSpecification(pkg_name, srv_name, request_message, response_message)
+
+
+def parse_action_file(pkg_name, interface_filename):
+    basename = os.path.basename(interface_filename)
+    action_name = os.path.splitext(basename)[0]
+    with open(interface_filename, 'r') as h:
+        return parse_action_string(pkg_name, action_name, h.read())
+
+
+def parse_action_string(pkg_name, action_name, action_string):
+    action_blocks = action_string.split(ACTION_REQUEST_RESPONSE_SEPARATOR)
```

Comment:
This logic is incorrect. It e.g. even matches a `---` in a comment. Please use the same logic as for splitting the request and response in a service.

Reply:
You're right. I had not foreseen the possibility of a separator in comments. So it's either above's logic using slices or `re.split('^---$', a, flags=re.MULTILINE)`.

Reply:
Fixed.